databricks_cli/stack/api.py

```diff
@@ -0,0 +1,336 @@
+            if RESOURCE_PHYSICAL_ID not in deployed_resource:
+                raise StackError("%s doesn't exist in deployed resource status" %
+                                 RESOURCE_PHYSICAL_ID)
+
+    def validate_config(self, stack_config):
+        """
+        Validate fields within a stack configuration. This ensures that an inputted configuration
+        has the necessary fields for stack deployment to function well.
+
+        :param stack_config: dict- stack config that is inputted by the user.
+        :return: None. Raises errors to stop deployment if there is a problem.
+        """
+        if STACK_NAME not in stack_config:
+            raise StackError("'%s' not in configuration" % STACK_NAME)
+        if STACK_RESOURCES not in stack_config:
+            raise StackError("'%s' not in configuration" % STACK_RESOURCES)
+        for resource in stack_config[STACK_RESOURCES]:
+            if RESOURCE_ID not in resource:
+                raise StackError("%s doesn't exist in resource config" % RESOURCE_ID)
+            if RESOURCE_SERVICE not in resource:
+                raise StackError("%s doesn't exist in resource config" % RESOURCE_SERVICE)
+            if RESOURCE_PROPERTIES not in resource:
+                raise StackError("%s doesn't exist in resource config" % RESOURCE_PROPERTIES)
+
+    def deploy_config(self, stack_config, stack_status=None):
+        self.validate_config(stack_config)
+        if stack_status:
+            self.validate_status(stack_status)
+            resource_id_to_status = self._get_resource_to_status_map(stack_status)
+        else:
+            resource_id_to_status = {}
+
+        stack_name = stack_config[STACK_NAME]
+        click.echo('Deploying stack %s' % stack_name)
+        resource_statuses = []
+
+        for resource_config in stack_config[STACK_RESOURCES]:
+            click.echo()
+            click.echo("Deploying resource")
+            # Retrieve resource deployment info from the last deployment.
+            resource_map_key = (resource_config[RESOURCE_ID], resource_config[RESOURCE_SERVICE])
+            resource_status = resource_id_to_status[resource_map_key] \
+                if resource_map_key in resource_id_to_status else None
+            # Deploy resource, get resource_status
+            new_resource_status = self.deploy_resource(resource_config, resource_status)
+            resource_statuses.append(new_resource_status)
+
+        # stack deploy status is original config with deployed resource statuses added
+        new_stack_status = stack_config
+        new_stack_status.update({STACK_DEPLOYED: resource_statuses})
+        new_stack_status.update({CLI_VERSION_KEY: CLI_VERSION})
+
+        # Validate that the status has been created correctly
+        self.validate_status(new_stack_status)
+
+        return new_stack_status
+
+    def deploy(self, config_path):  # overwrite to be added
```

Comment:
This method should be the first method in this class because it is the public API